torchvision/models/_utils.py

```diff
@@ -64,3 +73,395 @@ def forward(self, x):
+        assert len(set(available_nodes)) == len(available_nodes), \
+            "There are duplicate nodes! Please raise an issue https://github.com/pytorch/vision/issues"
+        # Check that all outputs in return_nodes are present in the model
+        for query in mode_return_nodes[mode].keys():
+            if not any([m.startswith(query) for m in available_nodes]):
+                raise ValueError(f"return_node: {query} is not present in model")
+
+        # Remove existing output nodes (train mode)
+        orig_output_nodes = []
+        for n in reversed(graph_module.graph.nodes):
+            if n.op == "output":
+                orig_output_nodes.append(n)
+        assert len(orig_output_nodes)
+        for n in orig_output_nodes:
+            graph_module.graph.erase_node(n)
+
+        # Find nodes corresponding to return_nodes and make them into output_nodes
+        nodes = [n for n in graph_module.graph.nodes]
+        output_nodes = OrderedDict()
+        for n in reversed(nodes):
+            if 'tensor_constant' in str(n):
+                # NOTE Without this control flow we would get a None value for
+                # `module_qualname = tracer.node_to_qualname.get(n)`.
+                # On the other hand, we can safely assume that we'll never need to
+                # get this as an interesting intermediate node.
+                continue
+            module_qualname = tracer.node_to_qualname.get(n)
+            for query in mode_return_nodes[mode]:
+                depth = query.count('.')
+                if '.'.join(module_qualname.split('.')[:depth + 1]) == query:
+                    output_nodes[mode_return_nodes[mode][query]] = n
+                    mode_return_nodes[mode].pop(query)
+                    break
+        output_nodes = OrderedDict(reversed(list(output_nodes.items())))
+
+        # And add them in the end of the graph
+        with graph_module.graph.inserting_after(nodes[-1]):
+            graph_module.graph.output(output_nodes)
+
+        # Remove unused modules / parameters
+        graph_module.graph.eliminate_dead_code()
+        graph_module.recompile()
+
+        # Keep track of the tracer and graph so we can choose the main one
+        tracers[mode] = tracer
+        graphs[mode] = graph
+
+    # Warn user if there are any discrepancies between the graphs of the
+    # train and eval modes
+    _warn_graph_differences(tracers['train'], tracers['eval'])
+
+    # Build the final graph module
+    graph_module = DualGraphModule(
+        model, graphs['train'], graphs['eval'], class_name=name)
+
+    # Keep non-parameter model properties for reference
+    for attr_str in model.__dir__():
+        attr = getattr(model, attr_str)
+        if (not attr_str.startswith('_')
+                and attr_str not in graph_module.__dir__()
+                and not ismethod(attr)
+                and not isinstance(attr, (nn.Module, nn.Parameter))):
+            setattr(graph_module, attr_str, attr)
```

Comment:
What are some of the use-cases that you had in mind for this part of the code?
We are not copying methods (which is ok), but for example if a particular family of `nn.Module` has methods like `no_weight_decay` or `forward_features`, they won't get propagated.

What should be the "guarantees" that we provide when copying some of the attributes?

Reply:
@fmassa The use case I had in mind was for when there are properties that hold useful information. For example `model.out_features` is a common one in the `timm` library. 

Hmm I'll think about propagating methods a little more. I guess my original worry was that the methods might make use of parameters, and not all parameters will necessarily exist in the final graph module.

Reply:
> What should be the "guarantees" that we provide when copying some of the attributes?

Maybe you saw, but I have this in the docstring:
>An attempt is made to keep all non-parametric properties of the original
    model, but existing properties of the constructed `GraphModule` are not
    overwritten.

If you read that coming in fresh would it be sufficient or would you still have questions?

Reply:
@alexander-soare 

> Hmm I'll think about propagating methods a little more. I guess my original worry was that the methods might make use of parameters, and not all parameters will necessarily exist in the final graph module.

I agree with you, I think it's better for now to avoid dealing with methods, as it can have many issues as you pointed out.
My question as more if we should restrict ourselves to a fixed set of properties to copy (like only numbers).

Also, in the case you pointed out, `model.out_features`, won't give the expected values for the new model, as we can extract features from arbitrary locations of the model (which can have a different number of channels).
This by itself could be a reason why we might not want to propagate meta-information, but I'm open for discussion.

(As a side note, I think we should stop relying on `out_features` for finetuning, and instead use `LazyModule` instead.

Reply:
I didn't think the `out_features` example through :sweat_smile: . I get you. I'm just going to drop this altogether actually. Better to leave the responsibility with the user if they want to propagate anything across, and we can learn from feature requests if there are any.

(didn't know about `LazyModule` - awesome!)